### DISCUSSION TITLE
Getting 409 conflict error in inertia response

### DISCUSSION THREAD

**Discussion by JensvandeWiel** [romsar/gonertia#6] [Q&A] [answered]
```
[18:21:56.112] WARN: Conflict {"error":{"code":0,"internal":null,"message":null},"id":"b1f693c3-4a7e-4931-9191-85e5fc3a4fb5","request":{"host":"localhost:8080","ip":"::1","length":0,"method":"POST","params":{},"path":"/log-in","query":"","referer":"http://localhost:8080/login","route":"/log-in"},"response":{"latency":140176,"length":0,"status":409}}
```

Handler:
```go
func (h *LoginInertiaHandler) ProcessLogin(ctx echo.Context) error {
	errCtx := gonertia.WithValidationError(ctx.Request().Context(), "email", "Invalid email address")
	req := ctx.Request().WithContext(errCtx)
	println(errCtx)
	i := middleware.GetInertia(ctx)
	i.Back(ctx.Response(), req)
	return nil
}
```

This results in a 409 conflict, am i doing anything wrong?

**Comment by JensvandeWiel** [comment]
I've learned that it only happens with i.Back()

**Comment by JensvandeWiel** [comment]
Also, this is how i make the request:
```tsx
const { data, setData, post, processing, errors,  } = useForm({
        email: '',
        password: '',
    });

    const handleSubmit = (e: MouseEvent) => {
        e.preventDefault();
        post('/login');
    }
```

**Reply by romsar** [reply]
Looks like it's normal.
If request was made by Inertia - Gonertia set status to 409 on any redirect responses just like in [official laravel adapter](https://github.com/inertiajs/inertia-laravel/blob/ddf47e999dcdabd2f370e5b9772a7449f4a81729/src/ResponseFactory.php#L110).
[This is important for core js package](https://github.com/inertiajs/inertia/blob/da297accbb78f9d14f766dbfdb04c517c05a7c5c/packages/core/src/router.ts#L201), so we will not change this behaviour.

**Reply by JensvandeWiel** [reply]
Yes i understand, but how would go back to the previous page and render it with errors included. I can just call the render function but if i make more complicated handlers that have props i will need to duplicate a lot of code if i want to add errors

**Comment by JensvandeWiel** [comment]
So what is the gonertia idiomatic way to just return a response with errors back to login?

**Comment by JensvandeWiel** [comment]
Inertia-laravel uses session to store the errors and then reloads the page

**Reply by romsar** [reply]
Yup, we don't have laravel framework benefits such as session manager.
I think the best way would be to write your own implementation or use a ready-made one, for example https://github.com/alexedwards/scs (flash messages are what you need).
I don't think the solution to this problem is the responsibility of this package, but if you have an idea, feel free to express it.

upd:
I thought a little and maybe we can accept some kind of interface with only `PushByUserID` (id, errors) and `PopByUserID` (id) methods in Gonertia's constructor.
These methods will save/receive validation errors for user in redirects. `PopByUserID` we can put in middleware, `PushByUserID` we can put in `gonertia.Location` method.
But, we need to think about how necessary it is or it is better not to take this responsibility.

**Reply by romsar** [reply]
Draft MR that I just made: https://github.com/romsar/gonertia/pull/7/files
What do you think?

**Reply by romsar** [reply]
I just realized that the Rails adapter uses a similar scheme: https://github.com/inertiajs/inertia-rails/blob/6e0fc5a92cafd5e821e6918168260d0e2fbce539/lib/inertia_rails/controller.rb#L67

**Reply by JensvandeWiel** [reply]
Yeah, im going to use sessions since i need to implement them for auth so...

**Reply by romsar** [reply]
Check version v1.1.8. Also I updated Readme file with simple inmem example.

**Comment by JensvandeWiel** [comment]
for now this works:
```go
func (h *LoginInertiaHandler) ProcessLogin(ctx echo.Context) error {
	errCtx := gonertia.WithValidationError(ctx.Request().Context(), "email", "Invalid email address")
	ctx.SetRequest(ctx.Request().WithContext(errCtx))
	return h.RenderLoginPage(ctx)
}
```
But this only works easily if the form endpoint is a subroute or the post version of the other route, since if it requires any parameters these should be available in the form submission endpoint too.

You can also use a session based system to pass errors

**Reply by romsar** [reply]
However, this is not a complete solution to the problem, I will merge https://github.com/romsar/gonertia/pull/7/files on this week, I think it will be useful.

**Comment by romsar** [comment, marked as answer]
Added `FlashProvider` in v1.1.8 version. Updated README.md.
